source/Preprocessor.cpp

```diff
@@ -472,6 +474,68 @@ Trivia Preprocessor::handleDefaultNetTypeDirective(Token directive) {
     return Trivia(TriviaKind::Directive, result);
 }
 
+Trivia Preprocessor::handleUndefDirective(Token directive) {
+    Token nameToken = expect(TokenKind::Identifier);
+    StringRef name = nameToken.valueText();
```

Comment:
good point , that would lead to confusing error messages. Will fix in both places.